src/appleseed/renderer/kernel/shading/shadingpoint.h

```diff
@@ -221,6 +221,7 @@ class ShadingPoint
     const Assembly& get_assembly() const;
 
     // Return the object instance that was hit.
+    ObjectInstance& get_object_instance();
```

Comment:
Can't you just change the existing `get_object_instance()` method to return a non-const ref but keeping the method itself const?